packages/sdk/src/marketplaces/LooksRare.ts

```diff
@@ -0,0 +1,573 @@
+    signature: string;
+    tokenId: string;
+    collection: string;
+    strategy: string;
+    currency: string;
+    signer: string;
+    isOrderAsk: boolean;
+    nonce: string;
+    amount: string;
+    price: string;
+    startTime: number;
+    endTime: number;
+    minPercentageToAsk: number;
+    params: unknown[];
+  }): Promise<MakeOrderResult> {
+    const apiKeyHeader: { "X-Looks-Api-Key": string } | {} = this.apiKey
+      ? {
+          "X-Looks-Api-Key": this.apiKey,
+        }
+      : {};
+
+    return fetch(this._getApiUrl(ApiPath.orders), {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+        ...apiKeyHeader,
+      },
+      body: JSON.stringify(payload),
+    })
+      .then((res) => res.json())
+      .then(({ success, data, message }: MakeOrderResponse) => {
+        if (!success) {
+          throw new Error(message);
+        }
+
+        if (!data) {
+          throw new Error("missing data");
+        }
+
+        return data;
+      });
+  }
+
+  /**
+   * Fetches order from LooksRare API.
+   */
+  private _fetchOrders(params: FetchOrderParams): Promise<MakeOrderResult[]> {
+    return fetch(this._getApiUrl(ApiPath.orders, parseFetchParams(params)))
+      .then((res) => res.json())
+      .then(({ success, data, message }: GetOrderResponse) => {
+        if (!success) {
+          throw new Error(message);
+        }
+
+        if (!data) {
+          throw new Error("missing data");
+        }
+
+        return data;
+      });
+  }
+
+  /**
+   * Gets LooksRare API origin.
+   */
+  private _getApiUrl(
+    path: ApiPath,
+    queryParams?: Record<string, string> | [string, string][]
+  ) {
+    const params = queryParams
+      ? "?" + new URLSearchParams(queryParams).toString()
+      : "";
+
+    return API_ORIGIN[this.chainId] + path + params;
+  }
+}
+
+/**
+ * Parses fetch params into 2D array format of [ [ key1, value1 ], [ key2, value2 ], ... ].
+ */
+function parseFetchParams(params: FetchOrderParams) {
+  const paramsArr: [string, string][] = [];
+
+  Object.entries(params).forEach(([key, value]) => {
+    if (Array.isArray(value)) {
+      return paramsArr.push(...flattenFetchParamArr(key, value));
+    }
+
+    if (typeof value === "object" && value !== null) {
+      return paramsArr.push(...flattenFetchParamObj(key, value));
+    }
+
+    paramsArr.push([key, value + ""]);
+  });
+
+  return paramsArr;
+}
+
+/**
+ * Flattens array named `A` with values of `[value1, value2]` into `[ [ 'A[]',  value1 ], [ 'A[]', value2 ] ]`.
+ */
+function flattenFetchParamArr(key: string, values: unknown[]) {
+  const flattened: [string, string][] = [];
+
+  for (const val of values) {
+    flattened.push([key + "[]", val + ""]);
+  }
+
+  return flattened;
+}
+
+/**
+ * Flattens object named `A` with value of `{ key: value }` into `[ 'A[key]', value ]`.
+ */
+function flattenFetchParamObj(objName: string, objValue: object) {
```

Comment:
// Not critical
Since eslint will force you to add return type, having `[string, string][]` in two places gonna look rough in my opinion. This method can be turned into:
```
function flattenFetchParamObj(objName: string, objValue: object): [string, string][] {
  return Object.entries(objValue).map(([key, value]) => {
    return [`${objName}[${key}]`, value];
  });
}
```

Same can be done with functions above.